src/app/services/users/users.service.ts

```diff
@@ -2,28 +2,27 @@ import {Injectable} from '@angular/core';
 import {User} from '../../../entities/user.model';
 import {Observable} from 'rxjs';
 import {AngularFirestore} from '@angular/fire/firestore';
+import { IFirebaseService } from '../firebase/models/firebase-service.interface';
+import { CollectionType } from '../firebase/models/collection-type.model';
+import { FirebaseService } from '../firebase/firebase.service';
 
 @Injectable()
-export class UsersService {
+export class UsersService implements IFirebaseService<User> {
+  collectionName: CollectionType = CollectionType.USERS;
 
-  constructor(private firestore: AngularFirestore) {
+  constructor(private firebaseService: FirebaseService) {}
+  
+  getAll$(): Observable<User[]> {
+    return this.firebaseService.getAllData$<User>(this.collectionName);
   }
-
-  getUserByID(id: string): Observable<User> {
-    return new Observable<User>(observer => {
-      this.firestore.collection<User>('users').doc(id).get().subscribe(user => {
-        observer.next(user.data() as User);
-      });
-    });
+  getById$(id: string): Observable<User> {
```

Comment:
Can firebaseservice be an abstract class thatimplements this functions for  a type T for it's users? Don't know if it works in type script.. but think about it. This way you only extend from firebaseService for the simple case.

Reply:
I would prefer that the services wouldn't know that they are speaking with a firebase service.
In order to do so I can't define the firebase service as an abstract service.